synapse/tests/utils.py

```diff
@@ -1451,62 +1428,69 @@ async def getTestCoreProxSvc(self, ssvc, ssvc_conf=None, core_conf=None):
 
                 yield core, prox, testsvc
 
+    @contextlib.contextmanager
+    def mayTestDir(self, dirn):
+
+        if dirn is not None:
+            yield dirn
+            return
+
+        with self.getTestDir() as dirn:
+            yield dirn
+
     @contextlib.asynccontextmanager
-    async def getTestAha(self, conf=None, dirn=None):
+    async def getTestAha(self, conf=None, dirn=None, ctor=None):
 
         if conf is None:
-            conf = {
-                'health:sysctl:checks': False,
-            }
-        conf = copy.deepcopy(conf)
+            conf = {}
+
+        if ctor is None:
+            ctor = s_aha.AhaCell.anit
+
+        conf = s_msgpack.deepcopy(conf)
+
+        hasdmonlisn = 'dmon:listen' in conf
+        hasprovlisn = 'provision:listen' in conf
+
+        network = conf.setdefault('aha:network', 'synapse')
+        hostname = conf.setdefault('dns:name', '00.aha.loop.vertex.link')
+
+        if hostname:
+            conf.setdefault('provision:listen', f'ssl://0.0.0.0:0?hostname={hostname}')
+            conf.setdefault('dmon:listen', f'ssl://0.0.0.0:0?hostname={hostname}&ca={network}')
+
+        conf.setdefault('health:sysctl:checks', False)
 
         with self.withNexusReplay():
-            if dirn:
-                async with await s_aha.AhaCell.anit(dirn, conf=conf) as aha:
-                    yield aha
-            else:
-                with self.getTestDir() as dirn:
-                    async with await s_aha.AhaCell.anit(dirn, conf=conf) as aha:
-                        yield aha
 
-    @contextlib.asynccontextmanager
-    async def getTestAhaProv(self, conf=None, dirn=None):
```

Comment:
This breaks downstream users

Reply:
I'm ok with this one. it's a unit test API that is easily replaced and completely vestigial.